packages/webpack-extensions/test/e2e/projects/metadata-plugin-project/webpack-before-emit.config.ts

```diff
@@ -0,0 +1,30 @@
+import { StylableMetadataPlugin } from '@stylable/webpack-extensions';
+import { StylableWebpackPlugin } from '@stylable/webpack-plugin';
+import { basename } from 'path';
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+
+module.exports = {
+    mode: 'development',
+    context: __dirname,
+    devtool: 'source-map',
+    plugins: [
+        new StylableWebpackPlugin({
+            optimize: {
+                shortNamespaces: true,
+            },
+        }),
+        new StylableMetadataPlugin({
+            name: 'test',
+            version: '1.0.0',
+            beforeEmit(builder) {
+                const metadata = builder.build();
```

Comment:
it's odd that a `beforeEmit` hook is the place when `build()` is called.

should it get the `metadata` from parameters? (without full access to the builder)